enclave_manager/avalon_enclave_manager/wpe/wpe_enclave_manager.py

```diff
@@ -94,53 +99,43 @@ def _manager_on_boot(self):
     def _process_work_order_sync(self, process_wo_id):
         """
         Process the work-order of the specified work-order id
-        and return the response.
-        Used for synchronous execution.
+        and return the response. Used for synchronous execution.
+
         Parameters:
-        - kv_helper is lmdb instance to access database
-        - process_wo_id is id of the work-order that is to be processed
+            @param process_wo_id - Id of the work-order that is to be processed
         """
-        logger.info("Processing work order")
-        try:
-            # Get all workorders requests from KV storage lookup and process
-            list_of_workorders = self._kv_helper.lookup("wo-scheduled")
-            if not list_of_workorders:
-                logger.info("Received empty list of work orders from " +
-                            "wo-scheduled table")
-                return
-        except Exception as e:
-            logger.error("Problem while getting ids from wo-scheduled table")
-            return
-
-        for wo_id in list_of_workorders:
-            if wo_id == process_wo_id:
-                resp = self._process_work_order_by_id(wo_id)
-                return resp
+        logger.info("Processing work orders found in wo-worker-pending table.")
+
+        # csv_match_pop will peek into the deque and pop out the first
+        # work-order id if it matches the process_wo_id
+        wo_id = self._kv_helper.csv_match_pop(
+            "wo-worker-pending", self._worker_id, process_wo_id)
+        while wo_id is not None:
+            if process_wo_id == wo_id:
+                return self._process_work_order_by_id(wo_id)
             else:
                 return None
-        # end of for loop
+            wo_id = self._kv_helper.csv_match_pop(
+                "wo-worker-pending", self._worker_id, process_wo_id)
+        # end of loop
+
+        logger.info("No more worker orders in wo-worker-pending table.")
 
 # -------------------------------------------------------------------------
 
     def _process_work_orders(self):
         """
         Executes Run time flow of enclave manager
         """
-        logger.info("Processing work orders")
-        try:
-            # Get all workorders requests from KV storage lookup and process
-            list_of_workorders = self._kv_helper.lookup("wo-scheduled")
-            if not list_of_workorders:
-                logger.info("Received empty list of work orders from " +
-                            "wo-scheduled table")
-                return
-        except Exception as e:
-            logger.error("Problem while getting keys from wo-scheduled table")
-            return
+        logger.info("Processing work orders found in wo-worker-pending table.")
 
-        for wo_id in list_of_workorders:
+        wo_id = self._kv_helper.csv_pop("wo-worker-pending", self._worker_id)
```

Comment:
WPE needs to check if it can support workload id present in the work order. Please add a TODO comment if you plan to do this in different PR

Reply:
Updated.